### DISCUSSION TITLE
Is new note creation triggered when plugin calls app.vault.create?

### DISCUSSION THREAD

**Discussion by farling42** [SilentVoid13/Templater#1262] [Help] [answered]
In relation to this issue (https://github.com/farling42/obsidian-import-json/issues/57) on my JSON importer module...

JSON Importer simply calls one of `app.vault.create` for a new note, or `app.vault.modify` or `app.vault.append` when updating an existing note.

Is there some other function which my module should call if the Templater plugin is enabled in order for the templates to be processed during note creation?

**Comment by Zachatoo** [comment, marked as answer]
Here's the pseudocode logic for how Templater decides whether or not to execute when a new file is created.

1. Trigger Templater on new file creation must be turned on.
2. Templater listens to the `app.vault.on("create")` event, then checks a few conditions (must meet **all** of the following conditions):
   1. Must be a markdown file.
   2. Must not be in the templates folder.
   3. The file must meet **one** of the following conditions:
      1. Must be empty on creation and folder templates setup for the folder the file was created in.
      2. The file's contents must be smaller than 100000 (see #873).

Hopefully that helps. Let me know if you need any more guidance.

**Reply by farling42** [reply]
Thanks for the rapid answer.
